components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/backchannellogout/DefaultLogoutTokenBuilder.java

```diff
@@ -326,9 +328,12 @@ private String getIdTokenIssuer(String tenantDomain) throws IdentityOAuth2Except
         if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
             try {
                 // Set the correct tenant domain before creating the path.
-                return ServiceURLBuilder.create().setTenant(tenantDomain).addPath(OAUTH2_TOKEN_EP_URL).build().
-                        getAbsolutePublicURL();
-            } catch (URLBuilderException e) {
+                ServiceURLBuilder serviceURLBuilder = ServiceURLBuilder.create().addPath(OAUTH2_TOKEN_EP_URL);
+                if (OrganizationManagementUtil.isOrganization(tenantDomain)) {
+                    return serviceURLBuilder.setOrganization(tenantDomain).build().getAbsolutePublicURL();
```

Comment:
If this is what set after /o/?
Then better to resolve organization id from tenant domain using the resolver available in organizationManager

There is a chance of having different value for tenant domain and org id